Opting into dependentupon convention

benvillalobos opened the pull request:
### Description

In SDK projects and new project system, we have eliminated the need for DependentUpon to make the appropriate file nesting in the IDE tree. However, there is a place where the build actually uses DependentUpon to:

1. Locate a source file, and _parse (!)_ it to get first class name and namespace
2. Generate .resources accordingly.

But people and features in VS rely on this and it has been a consistent source of feedback in moving to .NET Core 3.0. This opts into new behavior to use a convention instead of requiring explicit metadata.

Original issue: https://github.com/microsoft/msbuild/issues/4488

 This change is dependent upon: https://github.com/microsoft/msbuild/pull/4597

### Customer Impact

In a fairly common situation for projects that use resources, avoids the need to specify `DependentUpon` metadata for each resource.

### Regression?

No

### Risk

Low risk.

### Test changes in this PR
Added `DependentUponTest`. Currently it gets skipped, but this test was verified working on a local machine using a version of MSBuild that had the corresponding fix.

benvillalobos committed:
Opting into dependentupon convention

BenVillalobos commented:
Should this be conditional on target framework or used for anything using the sdk?

@nguerrera

nguerrera commented:
I think this should be conditional on target framework >= netcoreapp3.0 or >= netstandard2.1. Otherwise, this would be a potential breaking change.

nguerrera commented:
cc @davkean

benvillalobos committed:
Condition based on core 3.0 and framework 2.1

benvillalobos committed:
Created test for dependentupon

Note: This test fails without its MSBuild counterpart. I verified that this test ran successfully on a local machine with a version of MSBuild that had the needed change. Until that change is merged in, this test will skip & refer to the original issue.

wjk commented on src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:
```diff
@@ -95,6 +95,13 @@ Copyright (c) .NET Foundation. All rights reserved.
+    <EmbeddedResourceUseDependentUponConvention 
+      Condition="'$(EmbeddedResourceUseDependentUponConvention)' == '' and
+                 (('$(TargetFrameworkIdentifier)' == '.NETCoreApp' and '$(_TargetFrameworkVersionWithoutV)' >= '3.0') or
+                  ('$(TargetFrameworkIdentifier)' == '.NetFramework' and '$(_TargetFrameworkVersionWithoutV)' >= '2.1'))">true</EmbeddedResourceUseDependentUponConvention>
```
Typo? There is no .NET Framework 2.1, but there is a .NET Standard 2.1.

BenVillalobos replied:
Whoops!

nguerrera reviewed (changes requested):
Needs to be .netstandard 2.1 not .netframework.

benvillalobos committed:
Fixed framework identifier

benvillalobos committed:
Removed logging from test

Co-Authored-By: Rainer Sigwald <raines@microsoft.com>
